Assets/Programming/Scripts/UI/UpgradeMenuUI.cs

```diff
@@ -30,7 +30,9 @@ void Start()
     {
         _weapons = new List<Weapon>();
         _buttons = new List<Button>();
-        EventBroker.OnPlayerSpawned += PlayerSpawned;
+        //EventBroker.OnPlayerSpawned += PlayerSpawned; this was misbehaving so added the line below EP
+
+        _player = GameObject.FindGameObjectWithTag("Player");
```

Comment:
This is okay for now but I'd like to find a better solution for initializing the player. Using the if else in late update to do it works but It may be better practice to initialize it all in start. We'll likely try to debug the eventbroker problem and use that.